build/buildBuildImages.sh

```diff
@@ -89,18 +90,19 @@ function buildDockerImage() {
 	echo
 	echo Building a base image for tests...
 	# Do not write this image tag to the artifacts file as we do not intend to push it
-	docker build -t $dockerImageForTestsRepoName -f "$dockerFileForTestsToBuild" .
-	
-	echo "$dockerImageRepoName:latest" >> $ACR_BUILD_IMAGES_ARTIFACTS_FILE
+	testImageTag="$dockerImageForTestsRepoName:$dockerImageBaseTag"
+	docker build -t $testImageTag -f "$dockerFileForTestsToBuild" .
+
+	echo "$dockerImageRepoName:$dockerImageBaseTag" >> $ACR_BUILD_IMAGES_ARTIFACTS_FILE
 
 	# Retag build image with build number tags
 	if [ "$AGENT_BUILD" == "true" ]
 	then
-		uniqueTag="$BUILD_DEFINITIONNAME.$RELEASE_TAG_NAME"
+		uniqueTag="$dockerImageBaseTag-$BUILD_DEFINITIONNAME.$RELEASE_TAG_NAME"
```

Comment:
In the scenario of `oryx/build`, this would become `latest-Oryx-CI.20191105.2` which is incorrect.

Reply:
I tried looking at this PR's validation built output for building images step and I am seeing the following...Could you enable `set -ex` on this script as it can help us all see what the script is doing from the log.

```
List of images tagged (from '/data/agent/_work/84/s/artifacts/images/build-images-acr.txt'):
oryxdevmcr.azurecr.io/public/oryx/build:slim
oryxdevmcr.azurecr.io/public/oryx/build:slim-Oryx-Validation-GitHub.20191112.9
oryxdevmcr.azurecr.io/public/oryx/build:latest
oryxdevmcr.azurecr.io/public/oryx/build:latest-Oryx-Validation-GitHub.20191112.9 <------------
oryxdevmcr.azurecr.io/public/oryx/pack:Oryx-Validation-GitHub.20191112.9
```

Reply:
Updated this logic to use the base tag as a prefix if it's not `latest`.